packages/xod-client-electron/src/app/workspaceActions.js

```diff
@@ -91,11 +88,14 @@ export const getDefaultProjectPath = () => resolveDefaultProjectPath(getPathToBu
 const catchInvalidWorkspace = R.curry(
   (catchFn, err) => R.ifElse(
     R.compose(
-      isAmong([
-        FS_ERROR_CODES.INVALID_WORKSPACE_PATH,
-        FS_ERROR_CODES.WORKSPACE_DIR_NOT_EXIST_OR_EMPTY,
-        FS_ERROR_CODES.WORKSPACE_DIR_NOT_EMPTY,
-      ]),
+      R.both(
+        notNil,
+        isAmong([
+          FS_ERROR_CODES.INVALID_WORKSPACE_PATH,
+          FS_ERROR_CODES.WORKSPACE_DIR_NOT_EXIST_OR_EMPTY,
+          FS_ERROR_CODES.WORKSPACE_DIR_NOT_EMPTY,
+        ]),
+      ),
```

Comment:
:thinking: why doesn’t it work without the `notNil` check? I think it should. If there’s a problem, I think it’s a problem of `isAmong` implementation and should be fixed there.

Reply:
It's because hm-def rightfully complains that the thing we are trying to find has the wrong type. `[String] -> Null -> Boolean` does not satisfy`[a] -> a -> Boolean`. I'm not sure we should relax it to `[a] -> b -> Boolean`

Reply:
Error that passed here could not contain any `errorCode`, so it fails, cause signature `isAmong :: [a] -> a`, but got `[String] -> undefined`.
But initially, it was thought that any error will be accompanied by an error code, and therefore before this condition was not required. So in a perfect way, we should add error codes everywhere in new parts of pipeline... or easiest way — leave as is.